src/Spice86.Core/Emulator/CPU/CfgCpu/ParsedInstruction/Instructions/CallNearImm.cs

```diff
@@ -1,17 +1,24 @@
 namespace Spice86.Core.Emulator.CPU.CfgCpu.ParsedInstruction.Instructions;
 
 using Spice86.Core.Emulator.CPU.CfgCpu.InstructionExecutor;
+using Spice86.Core.Emulator.CPU.CfgCpu.InstructionRenderer;
 using Spice86.Core.Emulator.CPU.CfgCpu.ParsedInstruction.Instructions.CommonGrammar;
 using Spice86.Core.Emulator.CPU.CfgCpu.ParsedInstruction.Prefix;
 using Spice86.Shared.Emulator.Memory;
 
 public class CallNearImm : InstructionWithOffsetField<short> {
+    private readonly ushort _targetIp;
     public CallNearImm(SegmentedAddress address, InstructionField<ushort> opcodeField,
         List<InstructionPrefix> prefixes, InstructionField<short> offsetField) : base(address, opcodeField, prefixes,
         offsetField) {
+        _targetIp = (ushort)(NextInMemoryAddress.Offset + offsetField.Value);
     }
 
     public override void Execute(InstructionExecutionHelper helper) {
-        helper.NearCallOffset(this, OffsetField.Value);
+        helper.NearCallWithReturnIpNextInstruction(this, _targetIp);
+    }
+    
+    public override string ToAssemblyString(InstructionRendererHelper helper) {
+        return helper.ToAssemblyString("call near", helper.ToHex(_targetIp));
     }
```

Comment:
If you want syntax highlighting we're going to need something like this:
```suggestion
    public override void Format(IAssemblyFormatter formatter) {
        formatter.WriteMnemonic("call");
        formatter.WriteKeyword("near");
        formatter.WriteAddress(_targetIp);
    }
```
where a simple "StringFormatter" would just wrap a StringBuilder. While a highlighting one could create a list of string,color pairs, or an html string, or a sequence of Avalonia "Inlines", or whatever.
And the caller would be responsible for getting whatever they need out of the formatter.

Reply:
Let's go with an AST and a visitor that will generate the string. AST could be reused for execution in carefully compiled to an expression tree.

Reply:
Done in https://github.com/OpenRakis/Spice86/pull/1132